### PR TITLE
Add CI test for importability of bare Conda package

### LINKED ISSUE #1009: CI job to check for dependency leakage
Add a CI job to check that scipp can be imported when installed with no additional (optional) dependencies (e.g. matplotlib).

Re #1007

### PR DESCRIPTION
**Pull request by DanNixon**
Fixes #1009.

### CODE REVIEW

File: .azure-pipelines/pull_request.yml [hunk 1]
```diff
@@ -103,3 +103,4 @@ stages:
 
     jobs:
       - template: templates/documentation_build.yml
+      - template: templates/test_can_install_and_import.yml
```
Comment (SimonHeybrock) [inline .azure-pipelines/pull_request.yml]: Not necessary for the PR?

**Comment by DanNixon** [comment]
> Is there any check that it actually resolves to the latest package? I couldn't find it.

That isn't what is being tested here, and we also cannot really test that in a way that is useful.

File: .azure-pipelines/pull_request.yml (hunk 1 above)
Comment (DanNixon) [reply .azure-pipelines/pull_request.yml]: Why shouldn't it be tested for PRs?

Comment (SimonHeybrock) [reply .azure-pipelines/pull_request.yml]: I though of this as a safeguard against publishing bad packages, but unlikely to fail, so I thought we could safe the time.

Comment (DanNixon) [reply .azure-pipelines/pull_request.yml]: The time is small and is less than the documentation build which runs parallel to them.

We'd actually need it in PR builds to guard against publishing bad packages, by the time this fails on the `master` job the bad package has already been published.

**Pull request merged**